### PR TITLE
Fix for cloudsploit crash when action is * for iam:*

### PR DESCRIPTION
**Pull request by seswarrajan**
Signed-off-by: Eswar Rajan Subramanian [eswar@accuknox.com](mailto:eswar@accuknox.com)

### CODE REVIEW

File: plugins/aws/iam/rolePolicyUnusedServices.js [hunk 1]
```diff
@@ -412,6 +412,7 @@ module.exports = {
                                     if (statement.Action.length > 1 || statement.Action[0] !== '*') {
                                         for (let action of statement.Action) {
                                             if (config.whitelist_unused_actions_for_resources.includes(action)) continue;
+                                            if (action == "*") continue;
                                             let resourceAction = action.split(':')[1].toLowerCase();
 
                                             if (allServices[service] && !config.whitelist_unused_services.includes(service)) {
```
Comment (arnav13081994) [inline plugins/aws/iam/rolePolicyUnusedServices.js]: I think you should do 
```js
                                    if (statement.Action.length > 1 || statement.Action[0] !== '*') {
                                        for (let action of statement.Action) {
                                            if (config.whitelist_unused_actions_for_resources.includes(action) || 
                                                (action.length && action[0] === '*')) 
                                                continue;
                                            
                                            let resourceAction = action.split(':')[1].toLowerCase();


```

That is consistent with the coding style of aquasec

**Review by arnav13081994 — COMMENTED** [review]
Left a minor suggestion

**Review by arnav13081994 — APPROVED** [review]

**Review by sourabh-acn — APPROVED** [review]

File: plugins/aws/iam/rolePolicyUnusedServices.js [hunk 2]
```diff
@@ -411,7 +411,8 @@ module.exports = {
                                         statement.Action[0].split(':')[1].toLowerCase();
                                     if (statement.Action.length > 1 || statement.Action[0] !== '*') {
                                         for (let action of statement.Action) {
-                                            if (config.whitelist_unused_actions_for_resources.includes(action)) continue;
+                                            if (config.whitelist_unused_actions_for_resources.includes(action) ||
+                                                (action && action.length && action === '*')) continue;
```
Comment (nyrahul) [inline plugins/aws/iam/rolePolicyUnusedServices.js:415]: There are three equaltos! Is this correct?

**Review by nyrahul — CHANGES_REQUESTED** [review]

File: plugins/aws/iam/rolePolicyUnusedServices.js (hunk 2 above)
Comment (seswarrajan) [reply plugins/aws/iam/rolePolicyUnusedServices.js:415]: Yes @nyrahul 
== checks for value alone and === checks for type as wel.. The same condition is provided in line 403 as well. But missed here somehow.

Comment (seswarrajan) [reply plugins/aws/iam/rolePolicyUnusedServices.js:415]: Team. Please correct me if i am wrong

Comment (nyrahul) [reply plugins/aws/iam/rolePolicyUnusedServices.js:415]: Awesome! I didn;t knew about this at all. Thanks.

**Review by nyrahul — APPROVED** [review]
LGTM

**Pull request merged**